Xamarin.Forms.Platform.iOS/Renderers/RefreshViewRenderer.cs

```diff
@@ -0,0 +1,249 @@
+﻿using System;
+using System.ComponentModel;
+using UIKit;
+
+namespace Xamarin.Forms.Platform.iOS
+{
+	public class RefreshViewRenderer : ViewRenderer<RefreshView, UIView>
+	{
+		bool _isRefreshing;
+		bool _set;
+		nfloat _origininalY;
+		UIView _refreshControlParent;
+		UIRefreshControl _refreshControl;
+
+		public bool IsRefreshing
+		{
+			get { return _isRefreshing; }
+			set
+			{
+				bool changed = IsRefreshing != value;
+
+				if (changed)
+					TryOffsetRefresh(this, IsRefreshing);
+
+				_isRefreshing = value;
+
+				if (_isRefreshing)
+					_refreshControl.BeginRefreshing();
+				else
+					_refreshControl.EndRefreshing();
+			}
+		}
+
+		protected override void OnElementChanged(ElementChangedEventArgs<RefreshView> e)
+		{
+			base.OnElementChanged(e);
+
+			if (e.OldElement != null || Element == null)
+				return;
+
+			_refreshControl = new UIRefreshControl();
+
+			_refreshControl.ValueChanged += OnRefresh;
+
+			_refreshControlParent = this;
+
+			UpdateColors();
+			UpdateIsRefreshing();
+			UpdateIsEnabled();
+		}
+
+		protected override void OnElementPropertyChanged(object sender, PropertyChangedEventArgs e)
+		{
+			base.OnElementPropertyChanged(sender, e);
+
+			if (e.PropertyName == VisualElement.IsEnabledProperty.PropertyName)
+				UpdateIsEnabled();
+			else if (e.PropertyName == RefreshView.IsRefreshingProperty.PropertyName)
+				UpdateIsRefreshing();
+			else if (e.IsOneOf(RefreshView.RefreshColorProperty, VisualElement.BackgroundColorProperty))
+				UpdateColors();
+		}
+
+		protected override void SetBackgroundColor(Color color)
+		{
+			if (_refreshControl == null)
+				return;
+
+			if (color != Color.Default)
+				_refreshControl.BackgroundColor = color.ToUIColor();
+			else
+				_refreshControl.BackgroundColor = null;
+		}
+
+		protected override void Dispose(bool disposing)
+		{
+			base.Dispose(disposing);
+
+			if (disposing)
```

Comment:
reduced if nesting here. Also, introduce `_disposed` and set it true. See other renderers. We should use a similar pattern.